src/postgraphile.ts

```diff
@@ -0,0 +1,15 @@
+import { GlobalIdPluginOptions } from '.';
+
+// TODO: I couldn't get these to propagate up to projects that reference `@graphile/global-ids`
+
+declare module 'graphile-build' {
+  interface Options extends GlobalIdPluginOptions { }
+}
+
+declare module 'postgraphile-core' {
+  interface Options extends GlobalIdPluginOptions { }
+}
+
+declare module 'postgraphile' {
+  interface Options extends GlobalIdPluginOptions { }
+}
```

Comment:
These are all the same `Options` type; extending any one of them should work. I didn't know that you could add `extends` in declaration merging?

Options is currently defined as:

```
export interface Options {
  [str: string]: any;
}
```
https://github.com/graphile/graphile-engine/blob/ec3bcc496c00fd4b25293c06d2a890ffc234cf8b/packages/graphile-build/src/SchemaBuilder.d.ts#L20-L22

This needs fixing; but we need to finish converting core to TypeScript first. I'd not worry about this too much for now.

Reply:
> I didn't know that you could add extends in declaration merging?

Yea, these augmentations propagate to the `__tests__` scripts in this project, but even after augmenting everything up the chain, I couldn't get them to appear in a parent dependent project (at least when using `npm link @graphile/global-ids`).

Reply:
That's probably because you've got two different versions of graphile-build; one under your graphile/global-ids folder (devDependencies?) and one under your project.

Reply:
Ok, yea. So it may actually work in a proper `npm install` after we publish it.

Reply:
Maybe 🤷‍♂️ 

I solve this by having a convoluted dev setup with `rmlocal` scripts that ensure that everything uses the root dependencies of the graphile-engine project monorepo-style (even where the packages themselves aren't in the monorepo). https://github.com/graphile/postgraphile/blob/master/CONTRIBUTING.md

Reply:
Only one of these interfaces should remain; the `Options` interface is the same throughout (the one from graphile-build).